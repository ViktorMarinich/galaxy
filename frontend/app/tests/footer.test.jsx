var ReactTestUtils = require('react-addons-test-utils');
import React from 'react';
import renderer from 'react-test-renderer';
import { Link } from 'react-router';

it('renders correctly Footer', () => {
  const tree = renderer.create(
    <Footer id={0}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly footers logo', () => {
  const tree = renderer.create(
    <Footer id={3}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toContain("div");
  expect(tree.children[0].children[0]).toContain("Ant-eater");
  expect(tree.children[0].props.id).toContain("logo-footer");
});

//------------------------------------------------------------------------------

class Footer extends React.Component {
  render() {
    return (
      <div className="footer">
        <Link to={this.props.id=='0' ? "/" : "/profile" }
           data-remote="true" id="logo-footer">
           Ant-eater
        </Link>
      </div>
    );
  }
};
