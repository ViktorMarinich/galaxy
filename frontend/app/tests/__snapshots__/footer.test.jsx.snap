exports[`test renders correctly Footer 1`] = `
<div
  className="footer">
  <a
    data-remote="true"
    id="logo-footer"
    onClick={[Function]}
    style={Object {}}>
    Ant-eater
  </a>
</div>
`;

exports[`test renders correctly footers logo 1`] = `
<div
  className="footer">
  <a
    data-remote="true"
    id="logo-footer"
    onClick={[Function]}
    style={Object {}}>
    Ant-eater
  </a>
</div>
`;
